src/shared/conf-parser.c

```diff
@@ -495,8 +497,31 @@ static int config_parse_many_files(
                         return -ENOMEM;
         }
 
+        /* build an array of chased file paths */
+        STRV_FOREACH(fn, files) {
+                _cleanup_free_ char *chased = NULL;
+
+                r = chase(*fn, NULL, 0, &chased, NULL);
+                if (r < 0)
+                        continue;
+
+                r = strv_consume(&chased_files, TAKE_PTR(chased));
+                if (r < 0)
+                        return r;
+        }
+
         /* First read the first found main config file. */
         STRV_FOREACH(fn, conf_files) {
+                _cleanup_free_ char *chased = NULL;
+
+                r = chase(*fn, NULL, 0, &chased, NULL);
+                if (r >= 0) {
+                        if (strv_find(chased_files, chased)) {
+                                log_debug("Found symbolic link between config file and drop-ins. Skipping reading of config file");
```

Comment:
It's be nice to include the name of the file that we're skipping.
`log_debug("%s: symlink to drop-in, skipping reading.", fn);` or so.

Reply:
Or better:
`log_debug("%s: symlink to drop-in, will be read later.", fn);`